Source/ISHRemote/Trisoft.ISHRemote/Cmdlets/BackgroundTask/AddIshBackgroundTask.cs

```diff
@@ -159,24 +166,54 @@ protected override void EndProcessing()
             {
                 IshFields metadataFilter = new IshFields();
                 IshFields requestedMetadata = IshSession.IshTypeFieldSetup.ToIshRequestedMetadataFields(IshSession.DefaultRequestedMetadata, ISHType, new IshFields(), Enumerations.ActionMode.Find);
-                long progressId = 0;
+                var startEventResponse = new StartEventResponse();
+                var progressIds = new List<long>();
                 if (ParameterSetName == "IshObjectsGroup")
                 {
-                    var lngCardIds = _retrievedIshObjects.Select(ishObject => Convert.ToInt64(ishObject.ObjectRef[Enumerations.ReferenceType.Lng])).ToList();
-                    progressId = IshSession.DocumentObj25.RaiseEventByIshLngRefs(lngCardIds.ToArray(), EventType);
-                }
+                   var ishObjectsDividedInBatches = DevideListInBatchesByLogicalId(_retrievedIshObjects, IshSession.MetadataBatchSize);
 
+                    int currentIshObjectsCount = 0;
+                    foreach (var ishObjectsGroup in ishObjectsDividedInBatches)
+                    {
+                        // Create BackgroundTask
+                        var lngCardIds = ishObjectsGroup.Select(ishObject => Convert.ToInt64(ishObject.ObjectRef[Enumerations.ReferenceType.Lng])).ToList();
+                        var progressId = IshSession.DocumentObj25.RaiseEventByIshLngRefs(lngCardIds.ToArray(), EventType);
+                        progressIds.Add(progressId);
+                        currentIshObjectsCount += ishObjectsGroup.Count;
+                        WriteDebug($"RaiseEventByIshLngRefs.length[{lngCardIds.Count}] {currentIshObjectsCount}/{_retrievedIshObjects.Count}");
+                    }
+                }
+                
                 if (ParameterSetName == "ParameterGroup")
-                { 
+                {
                     // Start event 
                     var startEventRequest = new StartEventRequest
                     {
                         description = EventDescription,
                         eventType = EventType,
                         maximumProgress = _startEventMaxProgress
                     };
-                    var startEventResponse = IshSession.EventMonitor25.StartEvent(startEventRequest);
-                    
+                    startEventResponse = IshSession.EventMonitor25.StartEvent(startEventRequest);
+                }
+
+                if (ParameterSetName == "ParameterGroup" && StartAfter != null)
```

Comment:
For nullable types you can use HasValue. So StartAfter.HasValue instead of the check on null.